File: apps/frontend/services/connectionService.ts
import api from "./api";

export const connectionService = {
  sendRequest: async (receiverId: number): Promise<string> => {
    const res = await api.post<string>(`/api/connections/request/${receiverId}`);
    return res.data;
  },

  acceptRequest: async (senderId: number): Promise<string> => {
    const res = await api.post<string>(`/api/connections/accept/${senderId}`);
    return res.data;
  },

  rejectRequest: async (senderId: number): Promise<string> => {
    const res = await api.post<string>(`/api/connections/reject/${senderId}`);
    return res.data;
  },

  getMyConnections: async (): Promise<number[]> => {
    const res = await api.get<number[]>("/api/connections");
    return res.data;
  },

  checkConnection: async (otherUserId: number): Promise<boolean> => {
    const res = await api.get<boolean>(`/api/connections/check/${otherUserId}`);
    return res.data;
  },
};

File: apps/connection-service/src/main/java/com/joborbit/connection/service/ConnectionService.java
package com.joborbit.connection.service;

import java.util.List;

public interface ConnectionService {

    void sendRequest(Long senderId, Long receiverId);

    void acceptRequest(Long senderId, Long receiverId);

    void rejectRequest(Long senderId, Long receiverId);

    List<Long> getConnections(Long userId);

    boolean checkConnection(Long user1, Long user2);
}